pkg/platform/abstract/platform.go

```diff
@@ -263,6 +267,10 @@ func (ap *Platform) functionBuildRequired(createFunctionOptions *platform.Create
 		return true, nil
 	}
 
+	if createFunctionOptions.FunctionConfig.Spec.Build.CodeEntryType == s3EntryType {
+		return true, nil
```

Comment:
how is s3 is different from github / archive in that regard? why just `s3EntryType`?

Reply:
because they have the "Path" attribute filled and s3 doesn't